### PR TITLE
Rebase and 2 more commits for pages

### PR DESCRIPTION
**Pull request by hypeJunction**

### CODE REVIEW

**Comment by mrclay** [comment]
Can't merge due to conflicts.

**Comment by hypeJunction** [comment]
Try cherry-picking last two commits

File: mod/pages/start.php [hunk 1]
```diff
@@ -357,33 +358,40 @@ function pages_write_permission_check($hook, $entity_type, $returnvalue, $params
...
 	}
-	if ($page_guid = get_input('page_guid', 0)) {
+
+	if ($page_guid = get_input('page_guid')) {
 		$entity = get_entity($page_guid);
-	} elseif ($parent_guid = get_input('parent_guid', 0)) {
+	} elseif ($parent_guid = get_input('parent_guid')) {
 		$entity = get_entity($parent_guid);
 	}
-	if (isset($entity) && pages_is_page($entity)) {
```
Comment (mrclay) [inline mod/pages/start.php]: I don't understand why this last change is necessary. Will it work in 2.1?

Comment (hypeJunction) [inline mod/pages/start.php]: Ultimately canEdit = canWriteToContainer when it comes to adding new subpages to pages.

File: mod/pages/start.php [hunk 2]
```diff
@@ -382,6 +378,16 @@ function pages_container_permission_check($hook, $type, $return, $params) {
 		return;
 	}
 
+	$parent_container = $container->getContainerEntity();
+	if ($parent_container instanceof ElggGroup) {
+		// Only group members should be allowed to create subpages
+		return $parent_container->canWriteToContainer();
+	}
+
+	if (!elgg_in_context('pages')) {
```
Comment (hypeJunction) [inline mod/pages/start.php]: This logic is really bad and will fail during actions or in case of routed request.

File: mod/pages/start.php [hunk 3]
```diff
@@ -324,29 +325,29 @@ function pages_prepare_notification($hook, $type, $notification, $params) {
...
+				
 			case ACCESS_FRIENDS:
 				$owner = $entity->getOwnerEntity();
 				if (($owner instanceof ElggUser) && $owner->isFriendsWith($user->guid)) {
 					return true;
 				}
 				break;
 			default:
-				$list = get_access_array($user->guid);
+				$list = _elgg_services()->accessCollections->getAccessArray($user->guid);
```
Comment (hypeJunction) [inline mod/pages/start.php]: Need to make a public function for this, or?

**Pull request closed without merging**